genie/__main__.py

```diff
@@ -1,37 +1,100 @@
 #!/usr/bin/env python
-import argparse
 import genie
-
-
-def perform_main(syn, args):
-    if 'func' in args:
+import synapseclient
+import logging
+logger = logging.getLogger('genie')
+
+
+def synapse_login(username=None, password=None):
+    """
+    This function logs into synapse for you if credentials are saved.
+    If not saved, then user is prompted username and password.
+
+    :returns:     Synapseclient object
+    """
+    try:
+        syn = synapseclient.login(silent=True)
+    except Exception:
+        if username is None and password is None:
+            logger.info(
+                "Please specify --syn_user, --syn_pass the first time "
+                "you run this script.")
+        else:
+            syn = synapseclient.login(
+                email=username,
+                password=password,
+                rememberMe=True,
+                silent=True)
+    return(syn)
+
+
+def perform_validate(syn, args):
```

Comment:
Unsure why this genie-specific functionality got moved to the command line only?

Reply:
Honestly was just unsure what to name the function. I could name it main in validate.py but this is basically the function that invokes the validate.py functions.

Reply:
I wasn't worried about the name; but now there are two places with GENIE specific functionality (the command line tool and `validate.py`). If you capture it all in `validate.py` as it was it will be easier to turn that into an interface/class of some kind.

Reply:
Got it.  Will move it back, but keep `synapse_login` in `__main__`